pkg/reconciler/events/auditlogs/auditlogs_test.go

```diff
@@ -160,7 +160,9 @@ func TestAllCases(t *testing.T) {
 				WithCloudAuditLogsSourceServiceName(testServiceName),
 				WithCloudAuditLogsSourceAnnotations(map[string]string{
 					duckv1beta1.ClusterNameAnnotation: testingMetadataClient.FakeClusterName,
-				})),
+				}),
+				WithCloudAuditLogsSourceDefaultGCPAuth(),
```

Comment:
I think we should use `WithCloudAuditLogsSourceDefaultGCPAuth()` for every previous usage when we call `NewCloudAuditLogsSource()` I know in some cases, the UT can still pass but the logic is not correct.
The `Objects` should be a CloudAuditLogsSource with correct GCPAuth.
Similar for other resources.
I'd recommend you check usages for function `NewXXXX` and add `WithXXXDefaultGCPAuth` for all those usages.

Reply:
In addition, I found the the coding styles related to setting GCPAuth of different resources is not consistent. It will be better if you can fix the inconsistencies in this isuse